Grbl_Esp32/src/PinSettingsDefinitions.cpp

```diff
@@ -0,0 +1,960 @@
+#    define Y_PIN_PHASE_3 UNDEFINED_PIN
+#endif
+const char* Y_PIN_PHASE_3_DEFAULT = Y_PIN_PHASE_3;
+#ifndef Y2_PIN_PHASE_3
+#    define Y2_PIN_PHASE_3 UNDEFINED_PIN
+#endif
+const char* Y2_PIN_PHASE_3_DEFAULT = Y2_PIN_PHASE_3;
+#ifndef Z_PIN_PHASE_3
+#    define Z_PIN_PHASE_3 UNDEFINED_PIN
+#endif
+const char* Z_PIN_PHASE_3_DEFAULT = Z_PIN_PHASE_3;
+#ifndef Z2_PIN_PHASE_3
+#    define Z2_PIN_PHASE_3 UNDEFINED_PIN
+#endif
+const char* Z2_PIN_PHASE_3_DEFAULT = Z2_PIN_PHASE_3;
+#ifndef A_PIN_PHASE_3
+#    define A_PIN_PHASE_3 UNDEFINED_PIN
+#endif
+const char* A_PIN_PHASE_3_DEFAULT = A_PIN_PHASE_3;
+#ifndef A2_PIN_PHASE_3
+#    define A2_PIN_PHASE_3 UNDEFINED_PIN
+#endif
+const char* A2_PIN_PHASE_3_DEFAULT = A2_PIN_PHASE_3;
+#ifndef B_PIN_PHASE_3
+#    define B_PIN_PHASE_3 UNDEFINED_PIN
+#endif
+const char* B_PIN_PHASE_3_DEFAULT = B_PIN_PHASE_3;
+#ifndef B2_PIN_PHASE_3
+#    define B2_PIN_PHASE_3 UNDEFINED_PIN
+#endif
+const char* B2_PIN_PHASE_3_DEFAULT = B2_PIN_PHASE_3;
+#ifndef C_PIN_PHASE_3
+#    define C_PIN_PHASE_3 UNDEFINED_PIN
+#endif
+const char* C_PIN_PHASE_3_DEFAULT = C_PIN_PHASE_3;
+#ifndef C2_PIN_PHASE_3
+#    define C2_PIN_PHASE_3 UNDEFINED_PIN
+#endif
+const char* C2_PIN_PHASE_3_DEFAULT = C2_PIN_PHASE_3;
+
+// We need settings.h for the settings classes
+#include "Grbl.h"
+#include "Settings.h"
+#include "SettingsDefinitions.h"
+
+// Define the pins:
+
+PinSetting* CoolantFloodPin;       // COOLANT_FLOOD_PIN
+PinSetting* CoolantMistPin;        // COOLANT_MIST_PIN
+PinSetting* ProbePin;              // PROBE_PIN
+PinSetting* SDCardDetPin;          // SDCARD_DET_PIN
+PinSetting* SteppersDisablePin;    // STEPPERS_DISABLE_PIN
+PinSetting* StepperResetPin;       // STEPPER_RESET
+PinSetting* ControlSafetyDoorPin;  // CONTROL_SAFETY_DOOR_PIN
+PinSetting* ControlResetPin;       // CONTROL_RESET_PIN
+PinSetting* ControlFeedHoldPin;    // CONTROL_FEED_HOLD_PIN
+PinSetting* ControlCycleStartPin;  // CONTROL_CYCLE_START_PIN
+PinSetting* MacroButton0Pin;       // MACRO_BUTTON_0_PIN
+PinSetting* MacroButton1Pin;       // MACRO_BUTTON_1_PIN
+PinSetting* MacroButton2Pin;       // MACRO_BUTTON_2_PIN
+PinSetting* MacroButton3Pin;       // MACRO_BUTTON_3_PIN
+
+PinSetting* DynamixelTXDPin;  // DYNAMIXEL_TXD
+PinSetting* DynamixelRXDPin;  // DYNAMIXEL_RXD
+PinSetting* DynamixelRTSPin;  // DYNAMIXEL_RTS
+
+PinSetting* UserDigitalPin[4];
+PinSetting* UserAnalogPin[4];
+
+PinSetting* SpindleOutputPin;
+PinSetting* SpindleEnablePin;
+PinSetting* SpindleDirectionPin;
+PinSetting* SpindleForwardPin;
+PinSetting* SpindleReversePin;
+
+PinSetting* VFDRS485TXDPin;
+PinSetting* VFDRS485RXDPin;
+PinSetting* VFDRS485RTSPin;
+
+PinSetting* LimitPins[MAX_N_AXIS][2];
+PinSetting* StepPins[MAX_N_AXIS][2];
+PinSetting* DirectionPins[MAX_N_AXIS][2];
+PinSetting* DisablePins[MAX_N_AXIS][2];
+PinSetting* ClearToSendPins[MAX_N_AXIS][2];
+PinSetting* ServoPins[MAX_N_AXIS][2];
+PinSetting* StepStickMS3[MAX_N_AXIS][2];
+PinSetting* PhasePins[4][MAX_N_AXIS][2];
+
```

Comment:
PinSetting* spi_ssPin;
PinSetting* spi_sckPin;
PinSetting* spi_misoPin;
PinSetting* spi_mosiPin;

#include "Pin.h"

// Initialize the pin settings
void make_pin_settings() {
    spi_ssPin            = new PinSetting("SPI/SS/Pin", "GPIO.5");
    spi_sckPin           = new PinSetting("SPI/SCK/Pin", "GPIO.18");
    spi_misoPin          = new PinSetting("SPI/MISO/Pin", "GPIO.19");
    spi_mosiPin          = new PinSetting("SPI/MOSI/Pin", "GPIO.23");